source/_integrations/mqtt.markdown

```diff
@@ -559,44 +559,233 @@ The following software has built-in support for MQTT discovery:
 
 - Configuration topic: `homeassistant/binary_sensor/garden/config`
-- State topic: `homeassistant/binary_sensor/garden/state`
-- Payload:  `{"name": "garden", "device_class": "motion", "state_topic": "homeassistant/binary_sensor/garden/state"}`
+- Configuration payload:  `{"name": "garden", "device_class": "motion", "state_topic": "homeassistant/binary_sensor/garden/state"}`
 - Retain: The -r switch is added to retain the configuration topic in the broker. Without this, the sensor will not be available after Home Assistant restarts.
 
-To create a new sensor manually.
+To create a new sensor manually:
 
 ```bash
 mosquitto_pub -r -h 127.0.0.1 -p 1883 -t "homeassistant/binary_sensor/garden/config" -m '{"name": "garden", "device_class": "motion", "state_topic": "homeassistant/binary_sensor/garden/state"}'
 ```
 
-Update the state.
+Update the state:
 
 ```bash
 mosquitto_pub -h 127.0.0.1 -p 1883 -t "homeassistant/binary_sensor/garden/state" -m ON
 ```
 
-Delete the sensor by sending an empty message.
+Delete the sensor by sending an empty message:
 
  ```bash
 mosquitto_pub -h 127.0.0.1 -p 1883 -t "homeassistant/binary_sensor/garden/config" -m ''
 ```
 
-For more details please refer to the [MQTT testing section](/integrations/mqtt/#testing-your-setup).
+Note: the Configuration topic prefix must be `homeassistant` (or another if you changed the default). However, the State topic prefix does not need to be `homeassistant`. It is common for the State topic prefix to indicate which device or script sent the MQTT message, for example 'my_esp_device', 'rtl_433', or 'my_python_script'.
+
+#### Creating a device with two sensor entities (temperature and humidity sensor)
+
+In the following example, a device will be created with two sensor entities representing temperature and humidity measurements. Sensor entities can be represented by an [MQTT Sensor](/integrations/sensor.mqtt/).
+
+To set up a any device with multiple entities, a Configuration topic must be published for each entity. Each Configuration topic must include a "unique_id" property which is unique to the entity being created, and a "device" property which includes a device "identifiers" property which is unique to the device.
+
+A user, script, or device would send its configuration as JSON payload to the Configuration topic. After the first message to `config`, then the device may send MQTT messages with new sensor measurements in the form of a JSON payload.
+
+Set up a device with two sensor entities:
+
+- Configuration topic 1: `homeassistant/sensor/my_custom_sensor_001_temperature/config`
+- Configuration payload 1:
+```json
+{
+    "name": "My Custom Sensor 001 Temperature",
+    "unique_id": "my_custom_sensor_001_temperature",
+    "state_topic": "my_state_topic/my_custom_sensor/001",
+    "value_template": {% raw %} "{{ value_json.temperature }}" {% endraw %},
```

Comment:
Lets just add add unique_id and a shared device config to the current example, no need to replace the current ones.